packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -526,23 +526,38 @@ export class DatabaseCluster extends DatabaseClusterNew {
    * Adds the single user rotation of the master password to this cluster.
    */
   public addRotationSingleUser(options: RotationSingleUserOptions = {}): secretsmanager.SecretRotation {
+    if (options.secret) {
+      //TODO: check given secret is not master secret of this!
+      //TODO: check given secret is not added already!
+      //TODO: how to generate proper ID?
+      const id = 'AdditionalSecretRotation';
```

Comment:
I'm a little surprised that you used a different ID here. This means that this:

```ts
cluster.addRotationSingleUser();
cluster.addRotationSingleUser({ secret: mySecret });
```

will not fail. Is this correct? Can you have this sort of rotation on 2 different secrets in the same Cluster - will it work?

On a related note - does this work?

```ts
cluster.addRotationSingleUser({ secret: mySecret1 });
cluster.addRotationSingleUser({ secret: mySecret2 });
```

(I understand that with the code you've submitted, this will fail. But I'm asking whether this can work on the RDS level - can you have two rotating secrets like this attached to the same Cluster?)

Reply:
Yes, all the above are valid use cases. AFAIK you can have as many secret rotation as you want. The thing is, currently CDK provides single-user only for master credentials, and multi-user rotation for other credentials. This is not a fundamental limitation (it perfectly works as far as CFN is concerned). See the ticket for more info.

Reply:
If the above use-case is indeed viable:

```
cluster.addRotationSingleUser({ secret: mySecret1 });
cluster.addRotationSingleUser({ secret: mySecret2 });
```

then definitely the current code is not correct.

I think what we need to do is create the `SecretRotation` using the `secret` provided in the `options` as the scope, and not `this`, in the case when `RotationSingleUserOptions.secret` is not `undefined`.

Reply:
Right. Let me try it.

Reply:
OK, the type of `secret` is `ISecret` which cannot be used as the scope. Any idea?

Reply:
Cast it to `Construct`? 😃

Not very pretty, but should work here.

Reply:
I'm not sure how internally all this works. Can you confirm all `ISecret` instances can be cast to `Construct`?

Reply:
Yes.